### DISCUSSION TITLE
getState() is not working it returns an empty object.

### DISCUSSION THREAD

**Discussion by adityapatil9191** [data-driven-forms/react-forms#1460] [Need help] [answered]
Hello, I did not find any example who can help me with the correct way to get the form state without subscribing it . ( I want to avoid subscription because of the performance issues however I am open to both the approaches.) I did not find any example related to [getState](https://www.data-driven-forms.org/hooks/use-form-api#getstate). Any help will be really appreciated. @rvsia . Please find the snadbox below for reference. https://codesandbox.io/p/sandbox/react-data-driven-autocomplete-forked-9tkfvt?file=%2Fsrc%2FForm.jsx%3A17%2C26. In this example I want to log the entire state of the form at a particular moment. It should have all the values, formFields, errors etc present in it because I need to pass those values to another component.

**Comment by rvsia** [comment]
Hey, the link is not working, but using useFormApi should be simple. Are you calling the hook from the inside of the form?

**Reply by adityapatil9191** [reply]
I am calling the hook outside the form because I want to share the latest form state when the accordions open /close. I am using context API to do the same but the latest form state is not getting shared.not sure if I am doing anything wrong here. @rvsia

**Reply by rvsia** [reply, marked as answer]
You need to call it inside, otherwise there is no form context. The form renderer component creates and shares the context.

**Reply by adityapatil9191** [reply]
Okay thanks @rvsia . It worked

**Comment by adityapatil9191** [comment]
Access granted
